salt/_utils/kubernetes_utils.py

```diff
@@ -0,0 +1,829 @@
+"""Utility methods for manipulation of Kubernetes objects in Python.
+"""
+import datetime
+from functools import partial
+import inspect
+import keyword
+import operator
+from pprint import pformat
+import re
+
+from salt.ext import six
+from salt.utils.dictdiffer import recursive_diff
+
+try:
+    import kubernetes.config
+    import kubernetes.client as k8s_client
+    import kubernetes.client.apis as k8s_apis
+except ImportError:
+    HAS_LIBS = False
+else:
+    HAS_LIBS = True
+    ALL_APIS = frozenset(
+        api for _, api in inspect.getmembers(k8s_apis, inspect.isclass)
+    )
+
+__virtualname__ = 'metalk8s_kubernetes'
+
+
+def __virtual__():
+    if not HAS_LIBS:
+        return False, 'Missing dependencies: kubernetes'
+
+    return __virtualname__
+
+
+# Roughly equivalent to an Enum, for Python 2
+class ObjectScoped:
```

Comment:
Wasn't this named `ObjectScope` before? It represents _the scope of an object_, so it shouldn't be named _scope**d**_.

Reply:
Class was named `ObjectScope` and created with `ObjectScoped` (or the opposite) indeed I will change